src/test/lib/decorators_test.ts

```diff
@@ -389,6 +391,66 @@ suite('decorators', () => {
     });
   });
 
+  suite('@queryAssignedNodes', () => {
+    @customElement('assigned-nodes-el' as keyof HTMLElementTagNameMap)
+    class D extends LitElement {
+      @queryAssignedNodes() defaultAssigned!: Node[];
+
+      @queryAssignedNodes('footer', true) footerAssigned!: Node[];
+
+      render() {
+        return html`
+          <slot></slot>
+          <slot name="footer"></slot>
+        `;
+      }
+    }
+
+    @customElement(generateElementName() as keyof HTMLElementTagNameMap)
+    class C extends LitElement {
```

Comment:
Why do you need this element? To test slot forwarding?

Reply:
Yeah, to test `flatten`. Added comments to explain.